### ISSUE TITLE
nav_menu_link_attributes filter breaks third party menu plugins by replacing classes on all links

### ISSUE THREAD

**Issue by sevenspark** [brainstormforce/astra#1821]
Hi Astra Team,

In Astra v2.5 you've introduced this code in inc/markup-extras.php, line 887

```
function astra_menu_anchor_class_for_nav_menus( $atts ) {
	$atts['class'] = 'menu-link';
	return $atts;
}

add_filter( 'nav_menu_link_attributes', 'astra_menu_anchor_class_for_nav_menus' );
```

This filter *replaces* the class attribute on *every* menu item on the site with your "menu-link" class, breaking any non-Astra menus on the site.

The issue is that this code makes the assumption that every menu on the site is controlled by Astra.  As a result, replacing the classes on every menu item breaks other menu plugins, such as my own.  By replacing the class added by a custom walker, the CSS from the plugin no longer applies, breaking the display.

Please remedy this and make sure that your code only applies to your own menus.  For example, the nav_menu_link_attributes hook ( https://developer.wordpress.org/reference/hooks/nav_menu_link_attributes/ ) passes the $args array, so you can easily check that the menu being filtered is one being controlled by Astra (check the walker or container class or whatever defining feature).  For example:

```
function _astra_menu_anchor_class_for_nav_menus( $atts, $item, $args ) {
	if( isset( $args->menu_class ) && strpos( $args->menu_class, 'ast-nav-menu' ) !== false ){
		$atts['class'] = 'menu-link';
	}
	return $atts;
}
add_filter( 'nav_menu_link_attributes', '_astra_menu_anchor_class_for_nav_menus', 10, 3 );
```

In the meantime, anyone who comes across this can remove the problematic filter with the following code snippet:

```
function remove_astra_menu_link_class_filter(){
	remove_filter( 'nav_menu_link_attributes', 'astra_menu_anchor_class_for_nav_menus' );
}
add_action( 'wp', 'remove_astra_menu_link_class_filter' );
```

Thanks very much!

Chris

**Comment by SNKris** [comment]
@sevenspark your snippet saved me a great deal of headache, just in the nick of time. Thank you!

**Comment by sevenspark** [comment]
@SNKris glad it helped! :)

**Comment by premanshup** [comment]
@sevenspark  Thank you for reporting this issue, we are working on a fix and will soon release an update for this.

**Comment by realstephenzhao** [comment]
Yes, my webbsite Header menu created by Jetblock plugin just broke after I updated Astra theme to version 2.5.0

**Comment by shubhw12** [comment]
@sevenspark 
Thanks again for reporting this. We have released a quick fix ( v2.5.1) and the menu issues should be fixed now.
We really appreciate your concern for our product, I am closing this issue now.
